Assets/Scripts/Player/Body/Body.cs

```diff
@@ -101,5 +106,36 @@ private BodyPart FindBodyPart(BodyPartType bodyPartType)
 		{
 			return bodyParts.FirstOrDefault(bodyParts => bodyParts.BodyPartType == bodyPartType);
 		}
-	}
+
+        private void UpdateBodyPartList()
+        {
+            bodyParts.Clear();
+            foreach (BodyPart part in gameObject.GetComponentsInChildren<BodyPart>())
+            {
+                bodyParts.Add(part);
+                
+            }  
+        }
+
+        private void UpdateBodyPartVisuals(GameObject newBodyPart, BodyPart bodyPart)
+        {
+            // Making sure the part is equal to the original one
+            SkinnedMeshRenderer newBodyPartMesh = newBodyPart.GetComponent<SkinnedMeshRenderer>();
+            SkinnedMeshRenderer bodyPartMesh = bodyPart.SkinnedMeshRenderer;
+
+            Material[] materials = bodyPartMesh.sharedMaterials;
+            newBodyPartMesh.materials = materials;
+
+            // there's 5 shape keys on the human model
+            for (int i = 0; i < 4; i++)
```

Comment:
Any way to get the number dynamically? It would be a shame if merely updating the model required code changes.

Reply:
apparently, no.